lib/cobra_commander.rb

```diff
@@ -16,6 +17,10 @@ module CobraCommander
   UMBRELLA_APP_NAME = "App"
 
   def self.umbrella_tree(path)
-    ComponentTree.new(UMBRELLA_APP_NAME, path)
+    CalculatedComponentTree.new(UMBRELLA_APP_NAME, path)
+  end
+
+  def self.tree_from_cache(cache_file)
+    CachedComponentTree.from_cache_file(cache_file)
```

Comment:
any reason why we don't have just a simple constructor here? Instead of a static method that instantiates the object? I believe we can keep it as the `CalculatedComponentTree` and just do a `CachedComponentTree.new(cached_file)`. Or maybe two constructors, one that receives the path to a file, reads it and creates do object and other that expects the json.

Reply:
I'm not sure what you're suggesting.

Reply:
I'm suggesting that we replace the `tree_from_cached` method by another constructor, since it actually seems to be doing that already

Reply:
Could you maybe submit that change to make it clear?

Reply:
nevermind, I thought that ruby also had the concept of constructor overload but apparently it does not. Without it my suggestion does not make sense. I could still try something out, but not as simply.